File: semestr_3/aisd/study/2_selection_sort.py
def selection_sort(array):
    for i in range(len(array)):
        # Index of the decimal with the smalest value
        min_index = i
        for j in range(i+1,len(array)):
            if array[min_index] > array[j]:
                min_index = j
        tmp = array[i]
        array[i] = array[min_index]
        array[min_index] = tmp

# def selection_sort2(array):
#     for i in range(len(array)):
#         min_value = array[i]
#         min_index = i
#         for j in range(i+1,len(array)):
#             if min_value > array[j]:
#                 min_value = array[j]
#                 min_index = j
#         tmp = array[i]
#         array[i] = min_value
#         array[min_index] = tmp
    
arr = [5,3,2,4,1,6]
print(arr)
selection_sort(arr)
print(arr)

File: semestr_3/aisd/study/2_selection_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void wypisz(int *tab, int n){
    for(int i =0; i < n; i++)
        cout << tab[i] << ", ";
    cout << endl;
}

void selection_sort(int *tab, int n){
    int i, j, tmp, min_index;
    for(i = 0; i < n; i++){
        min_index = i;
        for(j = i+1; j < n; j++)
            if(tab[min_index] > tab[j])
                min_index = j;
        tmp = tab[i];
        tab[i] = tab[min_index];
        tab[min_index] = tmp;
    }
}

int main(){
    int arr[]{3,4,5,2,1};
    int arrSize = sizeof(arr)/ sizeof(arr[0]);
    wypisz(arr, arrSize);

    selection_sort(arr, arrSize);
    wypisz(arr, arrSize);
    return 0;
}
